Add developer mailing list for Shared Storage (#5302)
GoogleChrome/developer.chrome.com: site/en/_partials/privacy-sandbox/shared-storage-engage.md

in the future. If you try this API and have feedback, we'd love to hear it.

-  **GitHub**: Read the
   [proposal](https://github.com/WICG/shared-storage), [raise questions and participate in discussion](https://github.com/WICG/shared-storage/issues).

> -  **Shared Storage API announcements**: Join or view the mailing list at: https://groups.google.com/a/chromium.org/g/shared-storage-api-announcements

Reviewer: ```suggestion
-  **Shared Storage API announcements**: Join or view past announcements on our [mailing list](https://groups.google.com/a/chromium.org/g/shared-storage-api-announcements)
```

Reviewer: Links should always be stylized as such.
